tests/components/ViewedByCustomer/ViewedByCustomer.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { ViewedByCustomer } from '../../../src/Index';
+import { Mock, Fake } from 'coveo-search-ui-tests';
+import { IViewedByCustomerOptions } from '../../../src/components/ViewedByCustomer/ViewedByCustomer';
+import { IQueryResult } from 'coveo-search-ui';
+
+describe('ViewedByCustomer', () => {
+    describe('when the field isUserActionView of the result true', () => {
+        const fakeResult: IQueryResult = { ...Fake.createFakeResult(), isUserActionView: true };
+
+        it('should display the default label and the icon when no options are given', () => {
+            const testComponent = Mock.advancedResultComponentSetup<ViewedByCustomer>(ViewedByCustomer, fakeResult).cmp;
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-icon').length).toBe(1);
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-label').length).toBe(1);
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-label').item(0).innerHTML).toBe('Viewed by Customer');
+        });
+
+        it('should display the label given in the option if any is given', () => {
+            const option = new Mock.AdvancedComponentSetupOptions();
+            (option.cmpOptions as IViewedByCustomerOptions) = { label: 'test label' };
```

Comment:
const for you test label and also randomize it.

Reply:
I disagree: randomizing the label does not add value to the test and reduce the readability.
Same for the use of a const.

In my opinion:
The phases of a test (setup, exercise, verify, teardown) should be as loosely connected to each other, with the notable exception of setup and teardown, because they can sometimes be grouped up (3-phase test v.s. 4-phase test, same fight to be fair). In the present case, l19 is part of the exercise phase: We create the component following the instruction of the test (we put a custom label in the options). l22 is in the verify phase: We check that the label is the good one. If they were some treatment on the string, we would not want to reproduce it in the test, we'd want to see in just a quick sight what's going in, what's going out.

Reply:
I agree, don't randomize the value, but make sure it's not the default and that's enough. Randomizing a value adds nothing.

If you want to be really in the details, try with "dirty" strings to make sure it works.

Strings with É &; stuff <script> if you want to :)